test/development/basic/hmr.test.ts

```diff
@@ -16,467 +16,459 @@ import {
 import { createNext } from 'e2e-utils'
 import { NextInstance } from 'e2e-utils'
 import { outdent } from 'outdent'
-
-describe.each([['', '/docs']])(
-  'basic HMR, basePath: %p',
-  (basePath: string) => {
-    let next: NextInstance
-
-    beforeAll(async () => {
-      next = await createNext({
-        files: join(__dirname, 'hmr'),
-        nextConfig: {
-          basePath,
-        },
-      })
+import type { NextConfig } from 'next'
+
+describe.each([
+  { basePath: '', assetPrefix: '' },
+  { basePath: '', assetPrefix: undefined },
+  { basePath: '', assetPrefix: '/ap' },
+  { basePath: '/docs', assetPrefix: '' },
+  { basePath: '/docs', assetPrefix: undefined },
+  { basePath: '/docs', assetPrefix: '/ap' },
```

Comment:
```suggestion
  { basePath: '', assetPrefix: '' },
  { basePath: '', assetPrefix: '/asset-prefix },
  { basePath: '/docs', assetPrefix: '' },
  { basePath: '/docs', assetPrefix: '/ap' },
```

`undefined` values here aren't passed through to `next.config` as it uses `JSON.stringify` currently also this test suite is one of the longer ones to run so lets focus in the permutations we're checking.